### PR TITLE
tracer : http receiver pass header to TraceServiceServer.Export( ctx )

### PR DESCRIPTION
**Pull request by liguozhong**
## Important (read before submitting)
We are currently preparing for the upcoming 1.0 GA release. Pull requests that are not aligned with
the current roadmap https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/roadmap.md
and are not aimed at stabilizing and preparing the Collector for the release will not be accepted.

_Delete this paragraph before submitting._

**Description:** <Describe what has changed. 
Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.>
fix a tempo bug.
https://github.com/grafana/tempo/issues/495
https://github.com/grafana/tempo/issues/1381

**Link to tracking Issue:** <Issue number if applicable>
<https://github.com/open-telemetry/opentelemetry-collector/issues/5216>

**Testing:** < Describe what testing was performed and which tests were added.>

**Documentation:** < Describe the documentation added.>

_Please delete paragraphs that you did not use before submitting._

### CODE REVIEW

File: internal/data/opentelemetry-proto-gen/collector/trace/v1/trace_service.pb.gw.go [hunk 1]
```diff
@@ -59,7 +59,9 @@ func local_request_TraceService_Export_0(ctx context.Context, marshaler runtime.
 	if err := marshaler.NewDecoder(newReader()).Decode(&protoReq); err != nil && err != io.EOF {
 		return nil, metadata, status.Errorf(codes.InvalidArgument, "%v", err)
 	}
-
+	for key, val := range req.Header {
+		ctx = context.WithValue(ctx, key, val)
+	}
```
Comment (bogdandrutu) [inline internal/data/opentelemetry-proto-gen/collector/trace/v1/trace_service.pb.gw.go:64]: Read the first few lines of the file:

```
// Code generated by protoc-gen-grpc-gateway. DO NOT EDIT.
// source: opentelemetry/proto/collector/trace/v1/trace_service.proto
```

Comment (liguozhong) [reply internal/data/opentelemetry-proto-gen/collector/trace/v1/trace_service.pb.gw.go:64]: Thank you for review, I will study the code again and see how to correctly add the header to ctx

Comment (bogdandrutu) [reply internal/data/opentelemetry-proto-gen/collector/trace/v1/trace_service.pb.gw.go:64]: Also, I don't see this file in our repo anymore.

**Comment by bogdandrutu** [comment]
Closing since the file that is changed in this PR no longer exists, and we don't use the grpc gateway anymore.

**Pull request closed without merging**